module/mempool/stdmap/backdata/heropool/pool.go

```diff
@@ -0,0 +1,300 @@
+package heropool
+
+import (
+	"math/rand"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type EjectionMode string
+
+const (
+	RandomEjection = EjectionMode("random-ejection")
+	LRUEjection    = EjectionMode("lru-ejection")
+)
+
+// number of entries we try to eject before giving up and ejecting LRU
+const maximumRandomTrials = 10
+
+// EIndex is data type representing an entity index in Pool.
+type EIndex uint32
+
+// cachedEntity represents a cached entity that is maintained as a double linked list node.
+type cachedEntity struct {
+	owner  uint64
+	node   link
+	id     flow.Identifier
+	entity flow.Entity
+}
+
+type Pool struct {
+	size         uint32
+	free         state // keeps track of free slots.
+	used         state // keeps track of allocated slots to cachedEntities.
+	values       []cachedEntity
+	ejectionMode EjectionMode
+}
+
+func NewPool(limit uint32, ejectionMode EjectionMode) *Pool {
```

Comment:
Consider renaming `limit` to e.g. `preallocatedEntities` or something similar and intention revealing?

Reply:
`limit` has a well-known meaning across our codebase especially the `stdmap` package: It represents the maximum number of entities that are maintained in an in-memory pool. IMO it is more intention revealing and consistent than `preallocatedEntites`. Also, the suggested naming may add even more ambiguities around what it means by _preallocated_ entities. Examples of other places of `stdmap` package that we use the `limit` for the same purpose are [`Collections`](https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/module/mempool/stdmap/collections.go#L15), [`Guarantees`](https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/module/mempool/stdmap/guarantees.go#L16), [`Results`](https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/module/mempool/stdmap/results.go#L14), etc.

Reply:
> lso, the suggested naming may add even more ambiguities around what it means by preallocated entities. Examples of other places of stdmap package that we use the limit for the same purpose are Collections, Guarantees, Results, etc

Good point! :-)

Other LRU implementations such as Hashicorp call `limit` `size`. What about combining both naming worlds and renaming `limit` to `sizeLimit` which seems very intention revealing?

Reply:
https://github.com/onflow/flow-go/pull/1778/commits/ea6edb3818b6eda075a1f37e4a4f32e14b0fa2d1